addons/website/views/website_templates.xml

```diff
@@ -36,23 +36,37 @@
 
 <template id="layout" name="Main layout" inherit_id="portal.frontend_layout">
     <xpath expr="//html" position="before">
+        <!--
+            /!\ Notice that some `html_data` keys were moved between the `t-if`
+            version and the normal `t-set` below as a bug fix in stable 16.0.
+            Existing users before that fix may experience issues with:
+            - The publish button as a restrictor editor
+            - The "edit-in-backend" button as a restrictor editor
+            - The "Products" snippet as a visitor
+            - ...
+            None should be critical though (you just not have access to some
+            (important) features anymore). A update of the website app or
+            patching this view directly to have the following lines setting
+            `html_data` should solve the issue.
+        -->
         <t t-set="editable_in_backend" t-value="edit_in_backend or ('website_published' in main_object.fields_get() and main_object._name != 'website.page')"/>
         <t t-set="html_data" t-value="{
             'lang': lang and lang.replace('_', '-'),
             'data-website-id': website.id if website else None,
             'data-edit_translations': '1' if edit_translations else None,
+            'data-main-object': repr(main_object) if main_object else None,
+            'data-seo-object': repr(seo_object) if seo_object else None,
+            'data-is-published': 'website_published' in main_object.fields_get() and main_object.website_published,
+            'data-can-publish': 'can_publish' in main_object.fields_get() and main_object.can_publish,
+            'data-editable-in-backend': editable_in_backend,
```

Comment:
I'd prefer to not make this appear to people not restricted editor, at least especially for public user.

The other keys are not shown thanks to the `else None` which is not as easily applicable here.
Maybe you can find something in the qweb qcontext or add a key in `_prepare_frontend_environment` 'is_restricted_editor' ?

I'd at least find a way to hide editable-in-backend, can-publish? And is-published too if not used by snippets.